_drafts/cortex-m-fault-debug.md

```diff
@@ -0,0 +1,1079 @@
+
+For the Cortex M7, there is **no** way to force all stores to be synchronous / precise.
+
+#### Auxiliary Bus Fault Status Register (ABFSR) - 0xE000EFA8
+
+This register **only** exists for Cortex-M7 devices. When an `IMPRECISE` error occurs it will at least give you an indication of what memory bus the fault occurred on[^7]:
+
+![](img/cortex-m-fault/abfsr.png)
+
+A full discussion of memory interfaces is outside the scope of this article but more details can be found in the reference manual [^7].
+
+{: #mmfsr}
+
+### MemManage Status Register (MMFSR) - 0xE000ED28
+
+This register reports Memory Protection Unit faults.
+
+Typically MPU faults will only trigger if the MPU has been [configured and enabled](https://interrupt.memfault.com/blog/fix-bugs-and-secure-firmware-with-the-mpu) by the firmware. However, there are a few memory access errors that will always result in a MemManage fault -- such as trying to execute code from the system address range (`0xExxx.xxxx`).
+
+The layout of the register looks like this:
+
+![](img/cortex-m-fault/mmfsr.png)
+
+where,
+
+- `MMARVALID` - Indicates that the _MemManage Fault Address Register_ (**MMFAR**), a 32 bit register located at `0xE000ED34`, holds the address which triggered the MemManage fault.
+- `MLSPERR` & `MSTKERR` - Indicates that a MemManage fault occurred during lazy state preservation or exception entry, respectively. For example, this could happen if an MPU region is being used to detect [stack overflows](https://interrupt.memfault.com/blog/fix-bugs-and-secure-firmware-with-the-mpu#catch-stack-overflows-with-the-mpu).
+- `MUNSTKERR` - Indicates that a fault occurred while returning from an exception
+- `DACCVIOL` - Indicates that a data access triggered the MemManage fault.
+- `IACCVIOL` - Indicates that an attempt to execute an instruction triggered a MPU or Execute Never (XN) fault. We'll explore an example [below](#bad-pc-mpu-fault).
+
+## HardFault Status Register (HFSR) - 0xE000ED2C
+
+This registers explains the reason a HardFault exception was triggered.
+
+![](img/cortex-m-fault/hfsr.png)
+
+There's not too much information in this register but we will go over the fields real quickly
+
+- `DEBUGEVT` - Indicates that a debug event (i.e executing a breakpoint instruction) occurred while the debug subsystem was not enabled
+- `FORCED` - This means a configurable fault (i.e. the fault types we discussed in previous sections) was escalated to a hardfault, either because the configurable fault handler was not enabled or a fault occurred within the handler.
+- `VECTTBL` - Indicates a fault occurred because of an issue reading from an address in the vector table. This is pretty atypical but could happen if there is a bad address in the vector table and an unexpected interrupt fires.
+
+{: #registers-prior-to-exception}
+
+## Recovering Register State
```

Comment:
I'd name this something else. Like "Debugging Faults".

Reply:
went with "Recovering the Call Stack"